frontend/src/features/manageVehicles/hooks/useVehiclesApi.tsx

```diff
@@ -0,0 +1,30 @@
+import { IPowerUnit } from "../types/managevehicles";
+import { VEHICLES_API } from "./endpoints/endpoints";
+
+/*
+ *
+ * API Manager for the Manage Vehicles feature
+ *
+ */
+
+export const useVehiclesApi = () => {
+  /**
+   * Fetch*
+   All Power Unit Data
+   * @return {*}  {Promise<void>}
+   */
+  const getAllPowerUnits = async (): Promise<IPowerUnit[]> => {
+    const url = new URL(VEHICLES_API.GET_ALL_POWER_UNITS);
+
+    try {
+      const response = await fetch(url.href);
+      const json = await response.json();
+      return json;
+    } catch (error) {
+      console.error(error);
+      return [];
```

Comment:
I'd recommend removing this because, ideally, you want the consumer of this function to know the difference between an error and an actually empty list.

If you think we will revisit this when we iron out the error scenarios, then its fine to leave it this way.

Reply:
Agreed. I was planning on revisiting this next sprint